components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptDialogFragment.kt

```diff
@@ -25,3 +25,31 @@ internal abstract class PromptDialogFragment : DialogFragment() {
 
     val safeArguments get() = requireNotNull(arguments)
 }
+
+internal interface Prompter {
+
+    /**
+     * Invoked when a dialog is dismissed. This consumes the [PromptFeature]
+     * value from the session indicated by [sessionId].
+     *
+     * @param sessionId this is the id of the session which requested the prompt.
+     */
+    fun onCancel(sessionId: String)
+
+    /**
+     * Invoked when the user confirms the action on the dialog. This consumes
+     * the [PromptFeature] value from the session indicated by [sessionId].
+     *
+     * @param sessionId that requested to show the dialog.
+     * @param value an optional value provided by the dialog as a result of confirming the action.
+     */
+    fun onConfirm(sessionId: String, value: Any?)
```

Comment:
The `value` could be optional :)

This way we can avoid passing null parameter 
`feature?.onConfirm(sessionId, null)`

Reply:
Since it's rarely used, I prefer the more explicit `null` parameter here.

Reply:
👍